pkg/ebpf/controlplane/controller.go

```diff
@@ -192,6 +202,22 @@ func (ctrl *Controller) putSignalInPool(sig *signal) {
 	ctrl.signalPool.Put(sig)
 }
 
+// RegisterExternalSignal registers a signal handler for the given signal ID
+func (ctrl *Controller) RegisterExternalSignal(signalID events.ID, handler SignalHandler) {
```

Comment:
Can we have those signals be part of the extension interface?
Then, instead of registering new signals, we can iterate all registered extensions and get their signals, e.g. via new GetSignals() interface method